src/OrleansSQLUtils/Storage/RelationalOrleansQueries.cs

```diff
@@ -833,13 +833,26 @@ private static ReminderEntry CreateReminderEntry(IDataRecord record)
 
         private static IDbDataParameter CreateBeginHashParameter(IDbCommand command, uint beginHash)
         {            
-            return command.CreateParameter(ParameterDirection.Input, "beginHash", (int)beginHash);
+            return CreateHashRangeParameter(command, "beginHash", beginHash);
         }
 
 
         private static IDbDataParameter CreateEndHashParameter(IDbCommand command, uint endHash)
         {            
-            return command.CreateParameter(ParameterDirection.Input, "endHash", (int)endHash);
+            return CreateHashRangeParameter(command, "endHash", endHash);
+        }
+
+
+        private static IDbDataParameter CreateHashRangeParameter(IDbCommand command, string paramName, uint hash)
+        {
+            int intRange;
+            unchecked
+            {
+                const uint delta = (uint)int.MinValue;
```

Comment:
I believe Shay is on top of this, but just to facilitate discussion.

* The current, cast-only solution: `beginHash = 0 -> 0`, `endHash = 4294967295 -> -1`
* The fixed solution with overflow: `beginHash = 0 -> -2147483648 = int.MinValue`, `endHash = 4294967295 = 2147483647 = int.MaxValue`.